frontend/src/utils/encryption.ts

```diff
@@ -0,0 +1,16 @@
+const CryptoJS = require('react-native-crypto-js')
+
+const secretKey = process.env.EXPO_PUBLIC_ENCRYPTION_SECRET_KEY;
+console.log(secretKey)
+
+const encryptPassword = (password: string) => {
+  const encryptedPassword = CryptoJS.AES.encrypt(password, secretKey).toString();
+  return encryptedPassword;
+};
+
+const decryptPassword = (encryptedPassword: string) => {
+  const decryptedPassword = CryptoJS.AES.decrypt(encryptedPassword, secretKey).toString(CryptoJS.enc.Utf8);
+  return decryptedPassword;
+};
+
+export { encryptPassword, decryptPassword };
```

Comment:
I'd recommend utilising Node.JS' internal crypto library to avoid unnecessary requirements that may fall victim to third-party supply chain attacks.

The crypto module natively supports AES including the GCM block cipher and it's 256bit variant, as you can see here:
https://gist.github.com/rjz/15baffeab434b8125ca4d783f4116d81

Reply:
I'm getting error while using crypto library in react native.

![Yj1GNDPDEG.png](https://github.com/user-attachments/assets/83dce9ae-6ffd-491f-bd25-8e38209befb5)